HarpGuru/HarpFace/HarpCell/useQuizCycle/index.ts

```diff
@@ -0,0 +1,47 @@
+import { useGlobal, useDispatch } from 'reactn'
+import { useEffect } from 'react'
+import { IsActiveIds } from 'harpstrata'
+
+import { usePositionAnalysis } from '../usePositionAnalysis'
+import { YXCoord } from '../types'
+import { ExperienceModes } from '../../../helpers/setGlobalReactNState'
+import { usePrevious } from '../../../helpers'
+
+export const useQuizCycle = (yxCoord: YXCoord): void => {
+  const [activeExperienceMode] = useGlobal('activeExperienceMode')
+  const [quizQuestion] = useGlobal('quizQuestion')
+  const { thisIsActiveId } = usePositionAnalysis(yxCoord)
+  const previousIsActiveId = usePrevious(thisIsActiveId, IsActiveIds.Inactive)
+
+  const requestNextQuestion = useDispatch('requestNextQuestion')
+  const revealAnswer = useDispatch('revealAnswer')
+  useEffect(() => {
+    if (
+      thisIsActiveId === IsActiveIds.Active &&
+      previousIsActiveId === IsActiveIds.Inactive &&
+      activeExperienceMode === ExperienceModes.Quiz
+    ) {
+      const revealTimer = setTimeout(() => {
+        // TODO: why does the test fail if I don't test for
+        // requestNextQuestion first? Is it just because I
+        // haven't mocked that global reducer in the test?
```

Comment:
If this is just because the reducers haven't been mocked then it should be an easy fix. Otherwise, let's think further.

Reply:
It appears that this is no longer a problem. Perhaps that's changed since this logic was moved to a custom hook. If I added a test for this file then perhaps it would come up again, but as I've explained in a commit message in this pull request, I am taking measured risks with respect to testing in the interest of time which is becoming a significant factor now.